bin/saya/src/args/shard.rs

```diff
@@ -0,0 +1,69 @@
+use std::fmt::Display;
+use std::str::FromStr;
+
+use clap::builder::PossibleValue;
+use clap::{Args, ValueEnum};
+use katana_primitives::FieldElement;
+use saya_core::SayaMode;
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub struct SayaModeArg(pub SayaMode);
+
+#[derive(Debug, Args, Clone)]
+pub struct ShardOptions {
+    #[arg(help = "Choose either ephemeral or persistent saya mode.")]
+    #[arg(long = "mode")]
+    pub saya_mode: SayaModeArg,
+
+    #[arg(help = "Piltover contract address.")]
+    #[arg(long = "piltover")]
+    pub piltover: FieldElement,
+}
+
+impl Default for SayaModeArg {
+    fn default() -> Self {
+        SayaModeArg(SayaMode::Ephemeral)
+    }
+}
+
+impl ValueEnum for SayaModeArg {
+    fn value_variants<'a>() -> &'a [Self] {
+        &[
+            SayaModeArg(SayaMode::Ephemeral),
+            SayaModeArg(SayaMode::Persistent),
+            SayaModeArg(SayaMode::PersistentMerging),
```

Comment:
What is this one? `sovereign`?

Reply:
So the one using the merger program recursively should be `soverein`?

Reply:
Hum, then no. It's something unrelated. I wasn't sure why we had a persistent and persistent merging. Seems to be only at provable program layer, right?

Reply:
It changes the program flow in both merging and updating state. It is a different program, so it produces a different output. I think at this point `PersistentMerging` should be treated as a sort of legacy program, as it is still useful, but  `Persistent` is a newer, "sounder" way to prove state.

Reply:
Let's remove `PersistentMerging` if it's not used anymore?

Reply:
Removed